ee/clickhouse/queries/funnels/test/test_funnel_persons.py

```diff
@@ -302,20 +309,31 @@ def test_first_step_breakdowns(self):
                 "breakdown": "$browser",
             }
         )
-        results = ClickhouseFunnelPersons(filter, self.team)._exec_query()
+        _, results = ClickhouseFunnelActors(filter, self.team).get_actors()
+        results = cast(List[SerializedPerson], results)
 
-        self.assertCountEqual([val[0] for val in results], [person1.uuid, person2.uuid])
+        self.assertCountEqual([val["id"] for val in results], [person1.uuid, person2.uuid])
 
-        results = ClickhouseFunnelPersons(
+        _, results = ClickhouseFunnelActors(
             filter.with_data({"funnel_step_breakdown": "Chrome"}), self.team
-        )._exec_query()
+        ).get_actors()
+        results = cast(List[SerializedPerson], results)
```

Comment:
@hazzadous This was the casting that I had to do. Is there a better way to handle union types

Reply:
We could either do type narrowing (looks like in mypy that could look like [this](https://mypy.readthedocs.io/en/latest/type_narrowing.html#user-defined-type-guards))

Or return a structure that we can do narrowing on e.g. have a `actor_type` attribute.

Or we could return an Actor abstraction structure.

Or have separate methods for get_person_actors or get_group_actors, given that we likely know up front what we're querying for?

Note sure what's appropriate here.

Reply:
Thanks. Added a parent type that defines the shared keys